app/src/main/java/net/squanchy/tweets/view/TweetsAdapter.java

```diff
@@ -31,19 +30,27 @@ public TweetViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
 
     @Override
     public void onBindViewHolder(TweetViewHolder holder, int position) {
-        holder.updateWith(delegate.getItem(position));
+        holder.updateWith(tweetList.get(position));
     }
 
     @Override
     public int getItemCount() {
-        return delegate.getCount();
+        return tweetList.size();
     }
 
     public boolean isEmpty() {
-        return delegate.getCount() == 0;
+        return tweetList.isEmpty();
     }
 
-    public void refresh(TweetsPageView.TimelineLoadingCallback timelineLoadingCallback) {
-        delegate.refresh(timelineLoadingCallback);
+    public void updateWith(List<Tweet> tweets) {
+        if (tweets.isEmpty()) {
```

Comment:
This is not correct ⚠️  You should always replace the tweets list with the new one (make it not final and swap them, rather than mutating), otherwise we wouldn't be able to show the empty state for example when needed.